packages/data/tests/unit/services/navi-metadata-test.ts

```diff
@@ -0,0 +1,266 @@
+      metricFunctions.every(fn => fn instanceof MetricFunctionMetadataModel),
+      '`elideOne` `MetricFunctionsMetadataModel`s are loaded into the keg'
+    );
+    assert.deepEqual(metricFunctions.mapBy('id'), [], 'All `elideOne` metric functions are loaded');
+
+    assert.ok(this.service.loadedDataSources.has('elideOne'), '`elideOne` data source is loaded');
+  });
+
+  test('loadMetadata - default data source', async function(this: Context, assert) {
+    assert.notOk(this.service.loadedDataSources.has('bardOne'), '`bardOne` data source is initially not loaded');
+    await this.service.loadMetadata();
+    assert.ok(this.service.loadedDataSources.has('bardOne'), 'default data source is loaded');
+  });
+
+  test('loadMetadata - multiple calls', async function(this: Context, assert) {
+    assert.expect(1);
+
+    this.server.urlPrefix = `${config.navi.dataSources[0].uri}/v1`;
+    this.server.get('/tables', function() {
+      assert.ok(true, 'initial metadata load executes a request');
+      return { tables: [] };
+    });
+    await this.service.loadMetadata();
+
+    this.server.get('/tables', function() {
+      assert.notOk(true, 'after metadata is loaded, a fetch request is not executed');
+    });
+  });
+
+  test('all', async function(this: Context, assert) {
+    await this.service.loadMetadata({ dataSourceName: 'bardTwo' });
+    await this.service.loadMetadata({ dataSourceName: 'elideOne' });
+
+    const bardTwoTables = this.service.all('table', 'bardTwo');
+    assert.deepEqual(bardTwoTables.mapBy('id'), ['inventory'], '`all` returns all `elideOne` tables');
+
+    const elideOneTables = this.service.all('table', 'elideOne');
+    assert.deepEqual(elideOneTables.mapBy('id'), ['table0', 'table1'], '`all` return all `elideOne` tables');
+
+    const allTables = this.service.all('table');
+    assert.deepEqual(
+      allTables.mapBy('id'),
+      ['inventory', 'table0', 'table1'],
+      '`all` return all loaded tables when data source is not specified'
+    );
+    assert.ok(
+      allTables.every(table => table instanceof TableMetadataModel),
+      'All returns instances of `TableMetadataModel`s'
+    );
+
+    assert.throws(
+      () => this.service.all('table', 'notLoaded'),
+      /Metadata must have the requested data source loaded: notLoaded/,
+      '`all` throws an error if attempting to get all models from a data source that is not loaded'
+    );
+  });
+
+  test('getById', async function(this: Context, assert) {
+    await this.service.loadMetadata({ dataSourceName: 'elideOne' });
+    await this.service.loadMetadata({ dataSourceName: 'bardTwo' });
+
+    const metricOne = this.service.getById('metric', 'metric1', 'elideOne');
+    assert.ok(
+      metricOne instanceof MetricMetadataModel,
+      '`getById` returns a loaded instance of `MetricMetadataModel` when requesting `metric` type'
+    );
+    assert.equal(metricOne?.name, 'Metric 1', '`getById returns a metadata model given a type, id, & datasource');
```

Comment:
```suggestion
    assert.equal(metricOne.name, 'Metric 1', '`getById returns a metadata model given a type, id, & datasource');
```

Reply:
`metricOne` could be `undefined` if the id is invalid/missing

Reply:
Well since we don't expect it maybe `metricOne!.name` otherwise I guess it's fine this way